jupytext/cli.py

```diff
@@ -417,15 +429,24 @@ def jupytext_single_file(nb_file, args, log):
             msg += " Maybe you mean 'jupytext --sync {}' ?".format(args.set_formats)
         raise ValueError(msg)
 
-    nb_dest = args.output or (
-        None
-        if not args.output_format
-        else (
-            "-"
-            if nb_file == "-"
-            else full_path(base_path(nb_file, args.input_format), args.output_format)
-        )
-    )
+    nb_dest = None
+    if args.output:
+        nb_dest = args.output
+    elif args.output_format and nb_file == "-":
+        nb_dest = "-"
+    elif args.output_format:
+        try:
+            bp = base_path(nb_file, args.input_format)
+        except InconsistentPath:
+            if args.ignore_unmatched:
+                log(
+                    "[jupytext] Ignoring unmatched input "
+                    "path {} for format {}".format(nb_file, args.input_format)
+                )
+                return 0
+            else:
+                raise
+        nb_dest = full_path(bp, args.output_format)
```

Comment:
Refactored the logic here a bit to make it clearer (to me at least) what's going on